src/main/java/org/dice_research/opal/launuts/lau/ExcelParser.java

```diff
@@ -0,0 +1,190 @@
+package org.dice_research.opal.launuts.lau;
+
+import org.apache.commons.collections4.ListUtils;
+import org.apache.jena.ext.com.google.common.collect.Iterables;
+import org.apache.poi.ss.usermodel.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Generic Parser to parse all the sheet from the xlsx.
+ * Used instead of {@link LauCsvParser}.
+ *
+ * @author Vikrant Singh
+ */
+
+
+public class ExcelParser implements LauReaderInterface {
+
+    public static final String FILE_NAME = "EU-28-LAU-2019-NUTS-2016.xlsx";
+
+    List<LauContainer> lauContainerList = new LinkedList<LauContainer>();
+    private HashMap<String, HashMap<String, LauContainer>> index;
+
+    private Workbook workbook;
+    private Sheet sheet;
+    private Map<String, List<String>> getCodes = new HashMap<>();
+    private HashMap<String, Integer> getkeys = new HashMap<String, Integer>();
+    private HashMap<String, LauContainer> laucodeToLauContainerMap = new HashMap<String, LauContainer>();
+    private List<String> countryIds;
+    private boolean parsed = false;
+
+    @Override
+    public LauReaderInterface setLauSourceDirectory() throws IOException {
+
+        workbook = WorkbookFactory.create(new File(FILE_NAME));
+        return this;
+    }
+
+    @Override
+    public List<String> getCountryIds() throws LauReaderException {
+
+        if (!parsed)
+            parse();
```

Comment:
Instead of checking for parsed multiple times, can we create a constructor which will call the parse method and then no need to check for parsed?

Reply:
Done .